database/pgsql/migrations/00001_initial_schema.go

```diff
@@ -159,13 +159,18 @@ var (
 			`CREATE INDEX ON vulnerability(namespace_id, name);`,
 			`CREATE INDEX ON vulnerability(namespace_id);`,
 
+			// vulnerability_affected_feature is a de-normalized table to store
+			// the affected features in a independent place other than the
+			// feature table to reduce table lock issue, and makes it easier for
+			// decoupling updater and the Clair main logic.
 			`CREATE TABLE IF NOT EXISTS vulnerability_affected_feature (
 				id SERIAL PRIMARY KEY, 
 				vulnerability_id INT REFERENCES vulnerability ON DELETE CASCADE,
 				feature_name TEXT NOT NULL,
+				feature_type INT NOT NULL REFERENCES feature_type ON DELETE CASCADE,
```

Comment:
doesn't this mean if the last package of a particular kind gets deleted, so will that entry in the feature_type table?
e.g. if the last source package is deleted `source` ceases to be a package type

Reply:
I think feature_type here is the parent table, so it's the other way around

Reply:
https://www.postgresql.org/docs/9.5/ddl-constraints.html
> CASCADE specifies that when a referenced row is deleted, row(s) referencing it should be automatically deleted as well.

row in vulnerability_affected_feature is referencing the feature_type table.